File: FRONTEND CLIENT/HMS_CaseStudy_FrontEnd/src/services/StaffService.js
// StaffService.js
import axios from "axios";

const STAFF_API_BASE_URL = "http://localhost:8084/api/staff";

class StaffService {
  getAllStaff() {
    return axios.get(STAFF_API_BASE_URL);
  }

  getStaffById(id) {
    return axios.get(STAFF_API_BASE_URL + "/" + id);
  }

  addStaff(staff) {
    return axios.post(STAFF_API_BASE_URL, staff);
  }

  updateStaff(id, staff) {
    return axios.put(STAFF_API_BASE_URL + "/" + id, staff);
  }

  deleteStaff(id) {
    return axios.delete(STAFF_API_BASE_URL + "/" + id);
  }
}

export default new StaffService();

File: BACKEND SERVER/BackEnd/Staff-service/src/main/java/com/microservice/staff/service/StaffService.java
package com.microservice.staff.service;


import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

import com.microservice.staff.entity.Staff;

public interface StaffService {
    List<Staff> getAllStaff();

    Staff getStaffById(String id);

    Staff addStaff(@Valid @NotNull Staff staff);

    Staff updateStaff(@NotNull String id, @Valid @NotNull Staff staff);

    void deleteStaff(String id);
}
